pkg/distributor/distributor.go

```diff
@@ -991,6 +992,64 @@ func (d *Distributor) MetricsForLabelMatchers(ctx context.Context, from, through
 	return result, nil
 }
 
+func (d *Distributor) MetricsForLabelMatchersStream(ctx context.Context, from, through model.Time, matchers ...*labels.Matcher) ([]metric.Metric, error) {
+	replicationSet, err := d.GetIngestersForMetadata(ctx)
+	queryLimiter := limiter.QueryLimiterFromContextWithFallback(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := ingester_client.ToMetricsForLabelMatchersRequest(from, through, matchers)
+	if err != nil {
+		return nil, err
+	}
+	mutex := sync.Mutex{}
+	metrics := map[model.Fingerprint]model.Metric{}
+
+	_, err = d.ForReplicationSet(ctx, replicationSet, func(ctx context.Context, client ingester_client.IngesterClient) (interface{}, error) {
+		stream, err := client.MetricsForLabelMatchersStream(ctx, req)
```

Comment:
I assume we were using `queryIngesterStream` as example for this functino. In `queryIngesterStream` I see there is `defer stream.CLoseSend()` how come we don't need it here?